include/zephyr/drivers/stepper_motor_controller.h

```diff
@@ -0,0 +1,279 @@
+	const struct device *dev, enum micro_step_resolution resolution);
+
+static inline int32_t
+z_impl_stepper_motor_controller_set_micro_step_res(const struct device *dev,
+						   enum micro_step_resolution resolution)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_controller_set_micro_step_res(dev, resolution);
+}
+
+__syscall int32_t stepper_motor_controller_enable_velocity_mode(const struct device *dev,
+								enum direction dir);
+
+static inline int32_t z_impl_stepper_motor_controller_enable_velocity_mode(const struct device *dev,
+									   enum direction dir)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_controller_enable_velocity_mode(dev, dir);
+}
+
+__syscall int32_t stepper_motor_controller_get_stall_status(const struct device *dev,
+							    bool *stall_status);
+
+static inline int32_t z_impl_stepper_motor_controller_get_stall_status(const struct device *dev,
+								       bool *stall_status)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_get_stall_status(dev, stall_status);
+}
+
+__syscall int32_t stepper_motor_controller_enable_stall_detection(const struct device *dev,
+								  bool enable_stall_detection);
+
+static inline int32_t
+z_impl_stepper_motor_controller_enable_stall_detection(const struct device *dev,
+						       bool enable_stall_detection)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_controller_enable_stall_detection(dev, enable_stall_detection);
+}
+
+__syscall int32_t stepper_motor_controller_set_stall_detection_threshold(
+	const struct device *dev, int32_t stall_detection_threshold);
+
+static inline int32_t
+z_impl_stepper_motor_controller_set_stall_detection_threshold(const struct device *dev,
+							      int32_t stall_detection_threshold)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_controller_set_stall_detection_threshold(
+		dev, stall_detection_threshold);
+}
+
+__syscall int32_t stepper_motor_controller_set_reference(const struct device *dev,
+							 int32_t position);
+
+static inline int32_t z_impl_stepper_motor_controller_set_reference(const struct device *dev,
+								    int32_t position)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_controller_set_reference(dev, position);
+}
+
+__syscall int32_t stepper_motor_controller_get_position(const struct device *dev,
+							int32_t *position);
+
+static inline int32_t z_impl_stepper_motor_controller_get_position(const struct device *dev,
```

Comment:
**suggestion:** see comment below, should document what units steps is measured in, most likely full-steps since micro-steps cannot really be reached, yet explicit is better than to assume